packages/js/product-editor/src/hooks/use-woo-entity-prop.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import { useEntityProp } from '@wordpress/core-data';
+
+interface Metadata {
+	id?: number;
+	key: string;
+	value: string;
+}
+
+function useWooEntityProp(
+	property: string
+): [ string, ( value: string ) => void ] {
+	const isMeta = property.startsWith( 'meta_data.' );
+
+	const [ entityPropValue, setEntityPropValue ] = useEntityProp< string >(
+		'postType',
+		'product',
+		property
+	);
```

Comment:
I think you should only call this when `! isMeta`, as otherwise you are asking for a prop value that never will exist.

Reply:
AFAIK, React doesn't let you call hooks conditionally. That's the reason I left it like this, even though this should not be used when isMeta is false.